src/driver/linux_resource/syscall_aarch64.c

```diff
@@ -6,15 +6,191 @@
 #include <ci/efrm/debug_linux.h>
 #include <ci/efrm/syscall.h>
 
+#if 1
+#define TRAMP_DEBUG(x...) (void)0
+#else
+#define TRAMP_DEBUG(x...) EFRM_WARN(x)
+#endif
+
 void** efrm_syscall_table = NULL;
 EXPORT_SYMBOL(efrm_syscall_table);
 
 
-static typeof(aarch64_insn_decode_immediate) *aarch64_insn_decode_immediate_sym;
-static typeof(aarch64_insn_read) *aarch64_insn_read_sym;
-static typeof(aarch64_insn_extract_system_reg) *aarch64_insn_extract_system_reg_sym;
-static typeof(aarch64_get_branch_offset) *aarch64_get_branch_offset_sym;
-static typeof(aarch64_insn_adrp_get_offset) *aarch64_insn_adrp_get_offset_sym;
+/* The kernel contains some neat routines for doing AArch64 code inspection.
+ * Some of them are available as inline routines, but some are unfortunately
+ * not exported to modules.
+ * So we use efrm_find_ksym() to resolve these functions if it is available.
+ * Otherwise we copy their implementations here.
+ */
+#ifdef EFRM_HAVE_NEW_KALLSYMS
+static typeof(aarch64_insn_decode_immediate) *ci_aarch64_insn_decode_immediate;
+static typeof(aarch64_insn_read) *ci_aarch64_insn_read;
+static typeof(aarch64_insn_extract_system_reg) *ci_aarch64_insn_extract_system_reg;
+static typeof(aarch64_get_branch_offset) *ci_aarch64_get_branch_offset;
+static typeof(aarch64_insn_adrp_get_offset) *ci_aarch64_insn_adrp_get_offset;
+#else
+/* X-SPDX-Source-URL: https://git.kernel.org/pub/scm/linux/kernel/git/stable/linux.git */
+/* X-SPDX-Source-Tag: v5.15.83 */
+/* X-SPDX-Source-File: arch/arm64/kernel/patching.c */
+/* X-SPDX-Source-File: arch/arm64/lib/insn.c */
+/* X-SPDX-License-Identifier: GPL-2.0-only */
+/* X-SPDX-Comment: Functions we need to inspect aarch64 instructions for the
+ *                 case when efrm_find_ksym() is unavailable. "ci" prefix is
+ *                 added to all the functions. */
+#include <asm/patching.h>
```

Comment:
I think in that case we need to remove this pair of include's because the kernels that don't have the copied C files (patching.c and insn.c) also don't seem to have the corresponding H files. However, please check the above comment: maybe we want to have separate C and H files?

Build-time check (Linux 5.10.35 GCC 9 Ubuntu 22):
```
  CC [M]  /root/iteterev/onload_internal/build/aarch64_linux-5.10.35-2.0.0.0/src/driver/linux_resource/syscall_aarch64.o
/root/iteterev/onload_internal/src/driver/linux_resource/syscall_aarch64.c:40:10: fatal error: asm/patching.h: No such file or directory
   40 | #include <asm/patching.h>
      |          ^~~~~~~~~~~~~~~~
compilation terminated.
```

Reply:
Besides this compile error there are also runtime problems with linux 5.10, which need fixing.

Reply:
Removed those includes - we really do not need them.
Placed linux functions into separate c/h files.
Fixed the code to be compatible with linux-5.10.